src/x3f_process.c

```diff
@@ -91,11 +118,12 @@ static int get_black_level(x3f_t *x3f,
   pixels = 0;
   black_sum_sqdev = alloca(colors*sizeof(double));
   memset(black_sum_sqdev, 0, colors*sizeof(double));
-  pixels += sum_area_sqdev(x3f, "DarkShieldTop", image, rescale, colors,
-			   black_level, black_sum_sqdev);
-  if (use_bottom)
-    pixels += sum_area_sqdev(x3f, "DarkShieldBottom", image, rescale, colors,
-			     black_level, black_sum_sqdev);
+
+  for (i=0; i<4; i++)
+    if (use[i]) 
+      pixels += sum_area_sqdev(x3f, area[i], colors,
+			       black_level, black_sum_sqdev);
```

Comment:
In case debug printouts are to be kept, standard deviations should really be printed too.